core/src/main/scala/org/http4s/blaze/channel/nio2/ClientChannelFactory.scala

```diff
@@ -56,22 +56,28 @@ final class ClientChannelFactory(
       }
       val scheduledTimeout = scheduler.schedule(onTimeout, connectTimeout)
 
-      ch.connect(
-        remoteAddress,
-        null: Null,
-        new CompletionHandler[Void, Null] {
-          def failed(exc: Throwable, attachment: Null): Unit = {
-            p.tryFailure(exc)
-            scheduledTimeout.cancel()
-          }
+      try {
+        ch.connect(
+          remoteAddress,
+          null: Null,
+          new CompletionHandler[Void, Null] {
+            def failed(exc: Throwable, attachment: Null): Unit = {
+              p.tryFailure(exc)
+              scheduledTimeout.cancel()
+            }
 
-          def completed(result: Void, attachment: Null): Unit = {
-            channelOptions.applyToChannel(ch)
-            p.trySuccess(new ByteBufferHead(ch, bufferSize = bufferSize))
-            scheduledTimeout.cancel()
+            def completed(result: Void, attachment: Null): Unit = {
+              channelOptions.applyToChannel(ch)
+              p.trySuccess(new ByteBufferHead(ch, bufferSize = bufferSize))
+              scheduledTimeout.cancel()
+            }
           }
-        }
-      )
+        )
+      } catch {
+        case ex: IllegalArgumentException =>
+          ch.close()
```

Comment:
What if this throws?  I think we should catch and log here so we don't mask the original exception.

Reply:
Thanks for the review and good point about the exception handling! I have updated the code. I am not sure whether the logger is included correctly. It looks like it is done this way in other parts of the code.